src/MakePotCommand.php

```diff
@@ -704,20 +704,38 @@ protected function extract_strings() {
 			}
 
 			if ( ! $this->skip_theme_json ) {
-				// This will look for the top-level theme.json file, as well as
-				// any JSON file within the top-level styles/ directory.
-				ThemeJsonExtractor::fromDirectory(
+				JsonSchemaExtractor::fromDirectory(
 					$this->source,
 					$translations,
 					[
-						'schema'         => JsonSchemaExtractor::THEME_JSON_SOURCE,
-						'schemaFallback' => JsonSchemaExtractor::THEME_JSON_FALLBACK,
-						'include'        => $this->include,
-						'exclude'        => $this->exclude,
-						'extensions'     => [ 'json' ],
-						'addReferences'  => $this->location,
+						// Only look for theme.json files in any folder, nothing else.
+						'restrictFileNames' => [ 'theme.json' ],
+						'schema'            => JsonSchemaExtractor::THEME_JSON_SOURCE,
+						'schemaFallback'    => JsonSchemaExtractor::THEME_JSON_FALLBACK,
+						'include'           => $this->include,
+						'exclude'           => $this->exclude,
+						'extensions'        => [ 'json' ],
+						'addReferences'     => $this->location,
 					]
 				);
+
+				// Themes can have style variations in the top-level "styles" folder.
+				// They're like theme.json but can have any name.
+				if ( $is_theme ) {
+					JsonSchemaExtractor::fromDirectory(
+						$this->source,
+						$translations,
+						[
+							'restrictDirectories' => [ 'styles' ],
```

Comment:
The style variations within the `styles` folder can be organized in sub-folders ([example](https://github.com/WordPress/wordpress-develop/tree/trunk/src/wp-content/themes/twentytwentyfive/styles)). I presume this is already considered by this argument, but I haven't found it in the i18n-command codebase, so I can't read how it works. I wanted to raise this to make sure it works properly.

Reply:
I'll add a test for this 👍

Reply:
Test passes just fine